accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -94,20 +94,21 @@ library section below.
 
 ### Static errors
 
-We say that a type `T` is **nullable** if `Null <: T`.  This is equivalent to
-the syntactic criterion that `T` is any of:
+We say that a type `T` is **nullable** if `Null <: T` and not `T <: Object`.
+This is equivalent to the syntactic criterion that `T` is any of:
   - `Null`
   - `S?` for some `S`
   - `S*` for some `S` where `S` is nullable
   - `FutureOr<S>` for some `S` where `S` is nullable
   - `dynamic`
   - `void`
 
-We say that a type `T` is **non-nullable** if `T <: Object`.  This is equivalent
-to the syntactic criterion that `T` is any of:
-  - `Object`, `int`, `bool`, `Never`, `Function`
-  - Any function type
-  - Any class type or generic class type
+We say that a type `T` is **non-nullable** if `T <: Object`.
```

Comment:
Making `int*` definitely and potentially non-nullable.

That makes sense if being nullable is only used for restricting member access.

However, below it says:

> It is an error if a top level variable, static variable, or instance field with
potentially non-nullable type has no initializer expression and is not
initialized in a constructor via an initializing formal or an initializer list
entry, unless the variable or field is marked with the `late` modifier.

That would make `int x;` invalid in a non-migrated library because the type of the field is `int*` which is not nullable, and therefore potentially non-nullable.

I think we migth *need* the fourth category of types for `*` types: They are both definitely nullable and definitely non-nullable (and not potentially either).

If we use definite nullable/non-nullable to enable features, and potentially nullable/non-nullable to detect errors them, then that would allow `*` types to work everywhere, without error triggering an error.
(It's going to be tricky to get right, though).

Reply:
We don't rely on * types for error suppression in legacy libraries - it's simply the case that none of these new errors apply.

Reply:
OK, so the quoted text does not apply to legacy libraries at all. Then it may work. I'll have to think about it.